### ISSUE TITLE
Control node has unexpected movement behavior when rotated

### ISSUE THREAD

**Issue by Mousedotexecutable** [godotengine/godot#85069]
### Godot version

v4.1.3.stable.official

### System information

Fedora Linux 38 (Workstation Edition) - v4.1.3.stable.official - Compatibility (GLES3)

### Issue description

The control node has inconsistent movement behavior when it's rotated while having it's global position changed.
For example it might move diagonally, or it has higher speed than expected.

This only happens when it's global position variable gets modified.

It works as expected when rotation_degrees is set to zero or when the position variable gets updated.

The control node is supposed to move similar to how a sprite 2D moves when it's position gets changed.

This also happens with Godot 4.2 RC 1

### Steps to reproduce

Make a root scene (This can be any node) and add a Control node. For visualization add a Colored rect as the child of the Control node.

Attach a script, and have it move the Control node's global position downwards by one, every frame.
Code example below this line:
```extends Node

func _physics_process(delta: float) -> void:
	$Control.global_position.y += 1 * delta
```

Go back to the editor and set the Control node's rotation to 1 in the inspector.

Then run the scene.

Reproduction

### Minimal reproduction project
Open master.tscn and run

[ControlNodeBugReproduction.zip](https://github.com/godotengine/godot/files/13404261/ControlNodeBugReproduction.zip)

**Comment by Mousedotexecutable** [comment]
I can't seem to reproduce the bug in version 4.3 rc 1 anymore.

**Comment by kleonc** [comment]
Fixed by #89502 (since v4.3.dev5.official [89f70e98d]).
